lib/modules/manager/sbt/extract.ts

```diff
@@ -323,23 +373,123 @@ export function extractPackageFile(
 
   try {
     parsedResult = scala.query(content, query, {
```

Comment:
Why do you re-initialize `Ctx` over and over for each dependency? It would be a lot simpler do initialize it just once, and use it to access `globalVars`, `registryUrls`, `deps`, etc. instead of passing them as arguments. You can still have temp variables in there. See gradle manager for ref.